Packages/SmithyTestUtil/Sources/RequestTestUtil/HttpRequestTestBase.swift

```diff
@@ -87,6 +91,19 @@ open class HttpRequestTestBase: XCTestCase {
         return builder.build()
         
     }
+
+    func deconflictHost(host: String, resolvedHost: String?) -> String? {
+        var deconflictedHost: String? = nil
+        if !host.isEmpty,
+           let urlFromHost = ClientRuntime.URL(string: "http://\(host)"),
+           let parsedHost = urlFromHost.host {
+            deconflictedHost = parsedHost
+        }
+        if let resolvedHost = resolvedHost {
+            deconflictedHost = resolvedHost
+        }
+        return deconflictedHost
```

Comment:
do you know at this point that we are definitely going to return a String? like does it need to be optional here, your return value?